creator-node/src/snapbackSM.js

```diff
@@ -47,63 +58,308 @@ class SnapbackSM {
+  async monitorSecondarySync (syncWallet, primaryClockValue, secondaryUrl) {
+    // Monitor the sync status
+    let syncMonitoringRequestParameters = {
+      method: 'get',
+      baseURL: secondaryUrl,
+      url: `/sync_status/${syncWallet}`,
+      responseType: 'json'
+    }
+    let startTime = Date.now()
+    let syncAttemptCompleted = false
+    while (!syncAttemptCompleted) {
+      try {
+        let syncMonitoringResp = await axios(syncMonitoringRequestParameters)
+        let respData = syncMonitoringResp.data.data
+        this.log(`processSync ${syncWallet} secondary response: ${JSON.stringify(respData)}`)
+        // A success response does not necessarily guarantee completion, validate response data to confirm
+        if (respData.clockValue === primaryClockValue) {
+          syncAttemptCompleted = true
+          this.log(`processSync ${syncWallet} clockValue from secondary:${respData.clockValue}, primary:${primaryClockValue}`)
+        }
+      } catch (e) {
+        this.log(`processSync ${syncWallet} error querying sync_status: ${e}`)
+      }
+      if (Date.now() - startTime > MaxSyncMonitoringDurationInMs) {
+        this.log(`ERROR: processSync ${syncWallet} timeout for ${syncWallet}`)
+        syncAttemptCompleted = true
+      }
+      // 1s delay between retries
+      await utils.timeout(1000)
+    }
+  }
+
+  // Main sync queue job
+  async processSyncOperation (job) {
+    const syncRequestParameters = job.data.syncRequestParameters
+    let isValidSyncJobData = (
+      ('baseURL' in syncRequestParameters) &&
+      ('url' in syncRequestParameters) &&
+      ('method' in syncRequestParameters) &&
+      ('data' in syncRequestParameters)
+    )
+    if (!isValidSyncJobData) {
+      logger.error(`Invalid sync data found`)
+      logger.error(job.data)
+      return
+    }
+    const syncWallet = syncRequestParameters.data.wallet[0]
+    const primaryClockValue = await this.getUserPrimaryClockValue(syncWallet)
+    const secondaryUrl = syncRequestParameters.baseURL
+    this.log(`------------------Process SYNC | User ${syncWallet} | Target: ${secondaryUrl} ------------------`)
+
+    // Issue sync request to secondary
+    await axios(syncRequestParameters)
+
+    // Monitor the sync status
+    await this.monitorSecondarySync(syncWallet, primaryClockValue, secondaryUrl)
+
+    // Exit when sync status is computed
+    // Determine how many times to retry this operation
+    this.log('------------------END Process SYNC------------------')
+  }
+
+  // Function which ensures that state machine has been initialized correctly
+  // If not available on startup, subsequent state machine will attempt to initialize until success
+  async initializeNodeIdentityConfig () {
+    this.log(`Initializing SnapbackSM`)
+    this.log(`Retrieving spID for ${this.endpoint}`)
+    const recoveredSpID = await this.audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceProviderIdFromEndpoint(
```

Comment:
why not use `this.recoverSpID()` function?

Reply:
good point, worth reusing